### PR TITLE
Bump alloy [provider, rpc-types, serde, transport, rpc-client, transp…

### PR DESCRIPTION
**Pull request by m1stoyanov**
…ort-http] to 0.1.4, alloy-primitives to 0.7.7 and revm to 11.0.0

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,52 +1,52 @@
...
+tokio = {version = "1", features = ["rt-multi-thread"]}
 tracing = "0.1"
 
 url = "2"
 
 [dev-dependencies]
-alloy-rpc-client = { version = "0.1.2" }
-alloy-transport-http = { version = "0.1.2" }
+alloy-rpc-client = {version = "0.1.4"}
+alloy-transport-http = {version = "0.1.4"}
```
Comment (mattsse) [inline Cargo.toml:52]: let's replace with `= "0.1.4"`

File: Cargo.toml [hunk 2]
```diff
@@ -1,52 +1,52 @@
...
-alloy-primitives = "0.7.3"
-alloy-provider = { version = "0.1.2", default-features = false }
-alloy-rpc-types = { version = "0.1.2", features = ["eth"] }
-alloy-serde = { version = "0.1.2", default-features = false }
-alloy-transport = { version = "0.1.2", default-features = false }
+alloy-primitives = "0.7.7"
+alloy-provider = {version = "0.1.4", default-features = false}
+alloy-rpc-types = {version = "0.1.4", features = ["eth"]}
+alloy-serde = {version = "0.1.4", default-features = false}
+alloy-transport = {version = "0.1.4", default-features = false}
```
Comment (mattsse) [inline Cargo.toml:31]: I'm being very pedantic here but please keep the surrounding white space i `{}`

**Review by mattsse — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (m1stoyanov) [reply Cargo.toml]: Done :)

File: Cargo.toml (hunk 2 above)
Comment (m1stoyanov) [reply Cargo.toml]: Fixed all the white spaces my Code ate :)

File: src/backend.rs [hunk 3]
```diff
@@ -687,12 +687,10 @@ impl DatabaseRef for SharedBackend {
         })
     }
 
-    fn block_hash_ref(&self, number: U256) -> Result<B256, Self::Error> {
-        if number > U256::from(u64::MAX) {
+    fn block_hash_ref(&self, number: u64) -> Result<B256, Self::Error> {
+        if number > u64::MAX {
             return Ok(KECCAK_EMPTY);
         }
```
Comment (mattsse) [inline src/backend.rs:693]: this check is now redundant and can be removed entirely

**Review by mattsse — CHANGES_REQUESTED** [review]
last nit

File: src/backend.rs (hunk 3 above)
Comment (m1stoyanov) [reply src/backend.rs]: Completely missed that
Fixed

**Review by mattsse — APPROVED** [review]

**Pull request merged**